pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/SyntacticJavaTokenizerFactory.java

```diff
@@ -4,55 +4,24 @@
 
 package net.sourceforge.pmd.lang.java.ast;
 
-import java.util.ArrayList;
-import java.util.Iterator;
-import java.util.List;
-
-import net.sourceforge.pmd.lang.LanguageVersion;
 import net.sourceforge.pmd.lang.TokenManager;
 import net.sourceforge.pmd.lang.ast.impl.javacc.CharStream;
 import net.sourceforge.pmd.lang.ast.impl.javacc.JavaccToken;
-import net.sourceforge.pmd.lang.java.JavaLanguageModule;
-import net.sourceforge.pmd.lang.java.internal.JavaLanguageProperties;
 
 /**
  * Creates a tokenizer, that uses the syntactic grammar to provide context
  * for the tokenizer when reducing the input characters to tokens.
- * <p>This is required with JEP 430: String Templates</p>.
  *
- * @see <a href="https://openjdk.org/jeps/430">JEP 430: String Templates (Preview)</a>
+ * @deprecated This implementation has been superseded. It is not necessary to parse Java code in order to tokenize it.
  */
+@Deprecated
 public final class SyntacticJavaTokenizerFactory {
     private SyntacticJavaTokenizerFactory() {
         // factory class
     }
 
+    @Deprecated
     public static TokenManager<JavaccToken> createTokenizer(CharStream cs) {
-        final List<JavaccToken> tokenList = new ArrayList<>();
```

Comment:
We could leave this impl here, but just deprecate it and don't use it anymore.

Reply:
Done